system-addon/content-src/components/LinkMenu/LinkMenu.jsx

```diff
@@ -1,52 +1,98 @@
 const React = require("react");
 const {injectIntl} = require("react-intl");
 const ContextMenu = require("content-src/components/ContextMenu/ContextMenu");
+const {actionTypes, actionCreators} = require("common/Actions.jsm");
 
 class LinkMenu extends React.Component {
-  constructor(props) {
-    super(props);
-    this.getOptions = this.getOptions.bind(this);
+  getBookmarkStatus(site) {
+    return (site.bookmarkGuid ? {
+      id: "menu_action_remove_bookmark",
+      icon: "bookmark-remove",
+      action: actionTypes.DELETE_BOOKMARK_BY_ID,
```

Comment:
Not sure if this will be more confusing vs simpler, but the action can be a string "DELETE_BOOKMARK_BY_ID" and let the postprocessing do `let {action, data, id} = option; option.onClick = () => …({type: actionTypes[action], data})` ;)